### PR TITLE
Implementation of docker load command, load the tar ball directly into /var/lib/docker/graph

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan shishir.mahajan@redhat.com

### CODE REVIEW

File: graph/load.go [hunk 1]
```diff
@@ -91,25 +86,79 @@ func (s *TagStore) CmdLoad(job *engine.Job) engine.Status {
...
+					break
+				}
+			}
 		}
 
-		layer, err := os.Open(path.Join(tmpImageDir, "repo", address, "layer.tar"))
-		if err != nil {
-			log.Debugf("Error reading embedded tar", err)
-			return err
+		for i := 0; i < 3; i++ {
```
Comment (gdm85) [inline graph/load.go]: pardon, why is it 3?

Comment (shishir-a412ed) [inline graph/load.go]: because there are 3 entries in each image layer. And I only need two of them {JSON, layer.tar}

**Comment by jfrazelle** [comment]
drone is failing on `docker load`

**Comment by shishir-a412ed** [comment]
@jfrazelle I have removed the compression code (gzip) from the client side. It was anyways making it slower. Please check now if it works.

**Comment by rhatdan** [comment]
We worked on the gzip but it caused the loading of an image to be way slower in the most common case.  To see this run gzip -c image | gunzip -c image.new
It adds lots of time, with little benefit unless you were going over a WAN which is not going to be the common case for doing docker loads, I don't believe.

**Comment by unclejack** [comment]
@shishir-a412ed The drone.io build is still failing.

**Comment by crosbymichael** [comment]
Can you please rebase and look into the reason why drone is failing to run the integration tests?  Looks to be an issue only with this PR.

**Comment by shishir-a412ed** [comment]
HI @crosbymichael,

I have rebased it again, and now it is throwing some other error.
http://pastebin.com/cUVinGTU: Throwing error while trying to tag an image 'emptyfs'. I checked on registry.hub.docker.com and I couldn't find an image with that name.

**Comment by crosbymichael** [comment]
@shishir-a412ed this is an error caused by the changes in this PR.  Load and save are currently broken

**Comment by shishir-a412ed** [comment]
@crosbymichael I understand that, but I should be able to atleast pull the image from docker hub (for the test case it is failing).
`docker pull emptyfs:latest`. Or is `docker pull` also related to docker load/save ?

**Comment by crosbymichael** [comment]
The test is building and tagging an image that we are going to save/load.  We don't hit the registry for anything in the tests.  You should be able to see how the images is getting tagged from the failed test.

**Comment by shishir-a412ed** [comment]
@crosbymichael 
Please see pastebin.com to correlate to my comments.
http://pastebin.com/ycpHKjXu

1) ERROR 1: Not able to `docker load --input /docker-scratch.tar`. I am not able to find the input tar anywhere. I would need the input tar to test as why the load command is failing. Sorry I know its somewhere in the code, I am just not able to find it !

2) ERROR 2: All the tests are failing to inspect or tag this image {emptyfs}. I checked the code too, its just trying to tag the image and failing, because it doesn't find any image with this name. If you say the `docker build` is failing, I still don't see the correlation between `docker build` and `docker load` command ?

**Comment by shishir-a412ed** [comment]
@crosbymichael 
Hi I have found the tarball. I am working on the fix now.

**Comment by jlhawn** [comment]
Hooray for code removal! This is the change we discussed on IRC on Friday, right? Not storing the tar archive along with the extracted content should save a lot of space.

LGTM

**Comment by shishir-a412ed** [comment]
@jlhawn yes that's right ! I ran some `docker load` tests on my machine, it's working fine. Not sure if the drone is failing on the code or something else ?

**Comment by vbatts** [comment]
LGTM

**Pull request merged**